service/access.go

```diff
@@ -347,6 +363,25 @@ func (a *accessService) updateAccessToken(ctx context.Context, domain, role, pro
 	return at.(*AccessTokenResponse), err
 }
 
+func (a *accessService) storeTokenCache(key string, acd *accessCacheData, expTimeDelta time.Time, expTime *jwt.NumericDate) {
+	oldTokenCacheData, _ := a.tokenCache.Get(key)
+	a.tokenCache.SetWithExpire(key, *acd, expTime.Sub(expTimeDelta))
+	if oldTokenCacheData != (accessCacheData{}) {
+		oldTokenCacheSize := accessCacheMemoryUsage(&oldTokenCacheData)
+		a.memoryUsage += accessCacheMemoryUsage(acd) - oldTokenCacheSize
+		return
+	}
```

Comment:
use the `ok` bool to compute size for existing `accessCacheData{}` entry too
```suggestion
	oldTokenCacheData, ok := a.tokenCache.Get(key)
	a.tokenCache.SetWithExpire(key, *acd, expTime.Sub(expTimeDelta))
	if ok {
		oldTokenCacheSize := accessCacheMemoryUsage(&oldTokenCacheData)
		a.memoryUsage += accessCacheMemoryUsage(acd) - oldTokenCacheSize
		return
	}
```